shell/src/main/java/org/apache/kafka/shell/MetadataNodeManager.java

```diff
@@ -268,7 +269,7 @@ private void handleCommitImpl(MetadataRecordType type, ApiMessage message)
                             "Can't handle ConfigResource.Type " + record.resourceType());
                 }
                 DirectoryNode configDirectory = data.root.mkdirs("configs").
-                    mkdirs(typeString).mkdirs(record.resourceName());
+                    mkdirs(typeString).mkdirs(Utils.isBlank(record.resourceName()) ? "<default>" : record.resourceName());
```

Comment:
nit: `Utils.isBlank` will be true when `str == null || str.trim().isEmpty()`. Although I don't think the resource name will be null here, I think it'd better not to allow null resource name here. So, maybe:
```
record.resourceName().isBlank() ? "<default>" : record.resourceName()
```

Reply:
You are right, we should not allow a null value here, btw, I think `isBlank` is not right, we should use `isEmpty` here.

Reply:
Make sense!